### ISSUE TITLE
Running plugin with all listeners/extensions in plugin.xml

### ISSUE THREAD

**Issue by Yoskaldyr** [dkandalov/live-plugin#177]
Is it possible to run live plugin with all  listeners/extensions defined in plugin.xml (Or maybe run created zip file with full plugin inside the plugin folder)?

**Comment by dkandalov** [comment]
> Is it possible to run live plugin with all listeners/extensions defined in plugin.xml

At the moment live plugin just runs the code in `plugin.kts` (or `plugin.groovy`). In theory, nothing stops it from parsing `plugin.xml` and (un)load all the related listeners/extensions. I experimented with it to load actions in about 50 lines of Kotlin. It quickly becomes tricky and it essentially reimplements the functionality of IntelliJ plugin loader.

You can also try https://plugins.jetbrains.com/docs/intellij/dynamic-plugins.html with auto-reload https://plugins.jetbrains.com/docs/intellij/ide-development-instance.html#enabling-auto-reload I looked at it a couple years ago and it was restarting IDE anyway, but it might work better now 🤷 

> Or maybe run created zip file with full plugin inside the plugin folder

Not sure I understood this 🤔

**Comment by Yoskaldyr** [comment]
> Not sure I understood this 🤔

`Create Kotlin Plugin Zip` feature creates zip file with plugin that can be loaded like classic plugin. And before creating plugin.xml file can be edited (more listeners/extension is added etc). So it will be great if this zip file can be run directly from the Live Plugin

**Comment by dkandalov** [comment]
👍 `plugin.xml` by live plugin is only editable to specify plugin id, name and description. I haven't tried addition more listeners/extensions.

Do you want to `plugin.xml` because it's more obvious how to register listeners/extensions?

**Comment by Yoskaldyr** [comment]
> Do you want to `plugin.xml` because it's more obvious how to register listeners/extensions?

yes. But only because JB is still changing internal plugin API and hard to find examples how to extend something in realtime. But xml definition is staying the same long period of time.

For me approach of your addon is much-much better, especially for a small plugins. I can see result of my code changes almost instantly. And standard plugin creation workflow is just overkill for such small changes.

**Comment by Yoskaldyr** [comment]
I need custom idea plugins just for own small team productivity (and maybe for community if it get some interest). But not for creating BIG and COMPLEX paid plugins.
That's why your plugin is the best for such things👍
